### PR TITLE
CLN: Remove workarounds that are not relevant

### PR DESCRIPTION
**Pull request by bashtage**
Remove workarounds for NumPy, SciPy and Pandas that are
not relevant for minimum supported packages

### CODE REVIEW

File: statsmodels/compat/numpy.py [hunk 1]
```diff
@@ -87,244 +87,114 @@
...
-
-        >>> a = np.array([1, 2, 6, 4, 2, 3, 2])
-        >>> u, indices = np.unique(a, return_inverse=True)
-        >>> u
-        array([1, 2, 3, 4, 6])
-        >>> indices
-        array([0, 1, 4, 3, 1, 2, 1])
-        >>> u[indices]
-        array([1, 2, 6, 4, 2, 3, 2])
+def npc_unique(ar, return_index=False, return_inverse=False):
```
Comment (josef-pkt) [inline statsmodels/compat/numpy.py]: this can be competely deleted, was fixed in numpy 1.7

**Comment by josef-pkt** [comment]
according to the numpy mailing list discussion the np.unique regression was only in numpy 1.6.2. AFAICS it was fixed in the next release and should be correct for us since numpy 1.7.1
https://mail.python.org/pipermail/numpy-discussion/2014-March/069566.html

so all np.unique workarounds can be removed, AFAICS.

**Comment by josef-pkt** [comment]
based on a file search `from statsmodels.compat.numpy import nanmean` is only used in multivariate.pca and tests and could be replace by np.nanmean.

np_matrix_rank has a lot more usage across many models, and might not be worth a large scale rename

**Comment by bashtage** [comment]
Updated, so should be ready.

**Comment by josef-pkt** [comment]
looks good to me (I don't know much about the pandas versions)

I will merge when it comes back green

**Comment by josef-pkt** [comment]
all green LGTM complains about an old possible not initialized variable in the optimizer code

merging, Thank @bashtage

**Pull request merged**